platform/ios/src/MGLMapView.mm

```diff
@@ -974,6 +975,9 @@ - (void)setContentInset:(UIEdgeInsets)contentInset animated:(BOOL)animated
     {
         [self didUpdateLocationWithUserTrackingAnimated:animated];
     }
+    
+    // Compass, logo and attribution button constraints needs to be updated.
+    [self setNeedsUpdateConstraints];
```

Comment:
Not that I can find. `-setNeedsUpdateConstraints` will make sure a future call to `-updateConstraintsIfNeeded` calls `-updateConstraints` = no loop. If `-adjustContentInset` would get triggered in a loop, the loop is interrupted either because the previous `contentInset` is equal to the new or in case you're setting your own `contentInset`, it would get interrupted before that because of `automaticallyAdjustsScrollViewInsets=NO`